cmake/alpakaCommon.cmake

```diff
@@ -14,6 +14,49 @@ include(CMakePrintHelpers) # for easier printing of variables and properties
 #-------------------------------------------------------------------------------
 # Options.
 
+# Compiler options
+function(alpaka_compiler_option name description default)
+    if(NOT DEFINED name)
+        set(ALPAKA_${name} ${default} CACHE STRING "${description}")
+        set_property(CACHE ALPAKA_${name} PROPERTY STRINGS "DEFAULT;ON;OFF")
+    endif()
+endfunction()
+
+# Add parameter to a variable or target
+#
+# This method is automatically documenting all parameter added to the compiler into the variables
+# ALPAKA_COMPILER_OPTIONS_HOST, ALPAKA_COMPILER_OPTIONS_DEVICE, and ALPAKA_COMPILER_OPTIONS_HOST_DEVICE.
+#
+# scope - which compiler is effected: DEVICE, HOST, or HOST_DEVICE
+# operation - operation used to append given parameter: set, list, or target
+# name - name of the variable or target
+# ... - parameter to appended to the variable or target 'name'
+function(alpaka_set_compiler_options scope operation name)
+    if(scope STREQUAL HOST)
+        set(ALPAKA_COMPILER_OPTIONS_HOST ${ALPAKA_COMPILER_OPTIONS_HOST} ${ARGN} PARENT_SCOPE)
+    elseif(scope STREQUAL DEVICE)
+        set(ALPAKA_COMPILER_OPTIONS_DEVICE ${ALPAKA_COMPILER_OPTIONS_DEVICE} ${ARGN} PARENT_SCOPE)
+    elseif(scope STREQUAL HOST_DEVICE)
+        set(ALPAKA_COMPILER_OPTIONS_HOST_DEVICE ${ALPAKA_COMPILER_OPTIONS_HOST_DEVICE} ${ARGN} PARENT_SCOPE)
+    else()
+        message(FATAL_ERROR "alpaka_set_compiler_option 'scope' unknown, value must be 'HOST', 'DEVICE', or 'HOST_DEVICE'.")
+    endif()
+    if(operation STREQUAL list)
+        set(${name} ${${name}} ${ARGN} PARENT_SCOPE)
+    elseif(operation STREQUAL set)
+        foreach(arg IN LISTS ARGN)
+            set(tmp "${tmp} ${arg}")
+        endforeach()
+        set(${name} "${${name}} ${tmp}" PARENT_SCOPE)
```

Comment:
Since this also appends to the variable `${name}` I fail to understand why we need to `set` operation. Can we not handle this with `list` instead?

Reply:
A variable that is not a list is concatenated by spaces. A list is `;` separated.

Reply:
Fair enough.